src/ExpressionEvaluator/CSharp/Source/ExpressionCompiler/CompilationContext.cs

```diff
@@ -842,67 +795,88 @@ private static Imports BuildImports(CSharpCompilation compilation, ImmutableArra
+                        {
+                            continue;
                         }
+
+                        break;
+                    }
                     case ImportTargetKind.Namespace:
+                    {
+                        NameSyntax targetSyntax;
+                        if (!SyntaxHelpers.TryParseDottedName(importRecord.TargetString, out targetSyntax))
                         {
-                            Debug.Assert(target != null, string.Format("Namespace import string '{0}' has no target", importString));
+                            // DevDiv #999086: Some previous version of VS apparently generated type aliases as "UA{alias} T{alias-qualified type name}". 
+                            // Neither Roslyn nor Dev12 parses such imports.  However, Roslyn discards them, rather than interpreting them as "UA{alias}"
+                            // (which will rarely work and never be correct).
+                            Debug.WriteLine($"Import record '{importRecord}' has syntactically invalid target '{importRecord.TargetString}'");
+                            continue;
+                        }
 
-                            NameSyntax targetSyntax;
-                            if (!SyntaxHelpers.TryParseDottedName(target, out targetSyntax))
+                        NamespaceSymbol namespaceSymbol;
+                        var portableImportRecord = importRecord as PortableImportRecord;
+                        if (portableImportRecord != null)
+                        {
+                            var targetAssembly = portableImportRecord.GetTargetAssembly<ModuleSymbol, AssemblySymbol>(module, module.Module);
+                            if ((object)targetAssembly == null)
+                            {
+                                namespaceSymbol = BindNamespace(binder, targetSyntax);
+                            }
+                            else if (targetAssembly.IsMissing)
                             {
-                                // DevDiv #999086: Some previous version of VS apparently generated type aliases as "UA{alias} T{alias-qualified type name}". 
-                                // Neither Roslyn nor Dev12 parses such imports.  However, Roslyn discards them, rather than interpreting them as "UA{alias}"
-                                // (which will rarely work and never be correct).
-                                Debug.WriteLine("Import string '{0}' has syntactically invalid target '{1}'", importString, target);
+                                Debug.WriteLine($"Import record '{importRecord}' has invalid assembly reference '{targetAssembly.Identity}'");
                                 continue;
                             }
+                            else
+                            {
+                                namespaceSymbol = targetAssembly.GlobalNamespace;
```

Comment:
Could this code path be taken for the case targetASssembly == null as well, instead of going thru BindNamespace? Only the starting namespace would be Compilation.GlobalNamespace, no?